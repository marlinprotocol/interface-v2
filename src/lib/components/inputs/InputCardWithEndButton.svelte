<script lang="ts">
	import InputCard from '$lib/atoms/cards/InputCard.svelte';
	import Text from '$lib/atoms/texts/Text.svelte';
	import TooltipIcon from '$lib/atoms/tooltips/TooltipIcon.svelte';
	import type { InputCardVariant } from '$lib/types/componentTypes';

	export let styleClass = '';
	export let tooltipText = '';
	export let title: string;
	export let variant: InputCardVariant | undefined;
	export let label: string = '';
	export let showTitle: boolean = true;
	let innerWidth = 0;
</script>

<svelte:window bind:innerWidth />
<InputCard {variant} {styleClass}>
	{#if label}
		<p
			class="absolute left-7 top-[-1px] z-[1] bg-white font-poppins text-sm font-normal leading-[1px] text-[#030115]"
		>
			{label}
		</p>
	{/if}
	<div class="flex items-center justify-between">
		{#if showTitle}
			<div class="flex items-center gap-1">
				<Text variant="small" text={title} />
				{#if !!tooltipText}
					<TooltipIcon {tooltipText} />
				{/if}
			</div>
		{/if}
		<slot name="titleEndButton" />
	</div>
	<div class="flex items-center justify-between">
		<slot />
		<slot name="endInfoBox" />
		<slot name="endButton" />
	</div>
</InputCard>
