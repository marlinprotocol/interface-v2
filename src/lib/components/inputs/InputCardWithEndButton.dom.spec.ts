import { render } from '@testing-library/svelte';
import { describe, it, expect } from 'vitest';
import InputCardWithEndButton from './InputCardWithEndButton.svelte';
import html from '@playpilot/svelte-htm';

describe('InputCardWithEndButton', () => {
	it("render's the component properly", () => {
		const { container } = render(InputCardWithEndButton, {
			title: 'Test Title',
			variant: 'v2Input'
		});
		expect(container).toMatchSnapshot();
	});

	it('renders the title properly', () => {
		const { getByText } = render(InputCardWithEndButton, {
			title: 'Test Title',
			variant: 'v2Input'
		});

		// Check if the title is rendered
		const titleElement = getByText('Test Title');
		expect(titleElement).toBeTruthy();
	});

	it('renders the tooltip when tooltipText is provided', () => {
		// Render the component with a tooltip
		const { container } = render(InputCardWithEndButton, {
			tooltipText: 'Tooltip Information',
			title: 'Test Title',
			variant: 'v2Input'
		});
		expect(container.innerHTML).contain('Tooltip Information');
	});

	it('does not render the tooltip when tooltipText is not provided', () => {
		// Render the component without passing tooltipText
		const { queryByAltText } = render(InputCardWithEndButton, {
			title: 'Test Title',
			variant: 'v2Input'
		});

		// Check if the tooltip is not rendered
		expect(queryByAltText('Info')).not.toBeTruthy();
	});

	it('renders content in designated slot', async () => {
		// Render component with content passed to default slot
		const { findByText } = render(html`
            <${InputCardWithEndButton} title=${'Test Titme'}>
             <div>content</div>
             <div slot='titleEndButton'>titleEndButton</div>
             <div slot='endButton'>endButton</div>
            </${InputCardWithEndButton}>
          `);
		// Assert slot content is present
		expect(await findByText('content')).toBeTruthy();
		expect(await findByText('titleEndButton')).toBeTruthy();
		expect(await findByText('endButton')).toBeTruthy();
	});
});
