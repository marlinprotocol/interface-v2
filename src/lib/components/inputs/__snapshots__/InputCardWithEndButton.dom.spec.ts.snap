// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`InputCardWithEndButton > render's the component properly 1`] = `
<body>
  <div>
    <div
      class=" w-full rounded-lg bg-base-200 p-4"
      data-testid="input-card"
    >
      <div
        class="flex items-center justify-between"
      >
        <div
          class="flex items-center gap-1"
        >
          <div
            class=" text-sm font-normal"
            data-testid="text"
          >
            Test Title
          </div>
          <!--&lt;Text&gt;-->
           
        </div>
         
      </div>
       
      <div
        class="flex items-center justify-between"
      >
         
      </div>
    </div>
    <!--&lt;InputCard&gt;-->
    <!--&lt;InputCardWithEndButton&gt;-->
  </div>
</body>
`;
